### PR TITLE
refactor: remove specs for old versions of Terraform and Pulumi

### PR DESCRIPTION
**Pull request by edvald**
BREAKING CHANGE:

All but the two most recently supported versions of Pulumi and Terraform have been removed. If you've explicitly specified any of those versions, you will need to update your project configuration.

### CODE REVIEW

**Comment by stefreak** [comment]
I wonder how the tests could have passed, as we specifically refer to an old Terraform version in the tests, for eample in https://github.com/garden-io/garden/blob/a72edb77ef1720dfed6846c850a62d8df6e8f4f7/plugins/terraform/test/test-project/garden.yml#L13

See also #4381

Did we disable the plugins tests? 🤔

**Comment by stefreak** [comment]
Yes, plugins tests are still disabled:

https://github.com/garden-io/garden/blob/main/.circleci/config.yml#L1030

File: core/src/server/server.ts [hunk 1]
```diff
@@ -712,6 +713,29 @@ export class GardenServer extends EventEmitter {
           })
         )
       }
+    } else if (requestType === "autocomplete") {
```
Comment (stefreak) [inline core/src/server/server.ts:716]: @edvald Did the autocomplite commits slip in accidentally to this PR, or is it purposefully part of this PR?

**Comment by Walther** [comment]
note: we are addressing some issues currently in a WIP branch here https://github.com/garden-io/garden/pull/4381
let's hold off on merging this PR before that one is done to avoid conflicts and issues

**Comment by vvagaytsev** [comment]
@Walther @stefreak @edvald it looks like we have to postpone this, right?

**Comment by edvald** [comment]
Closing this in favor of a more generic approach (basically allowing _any_ version to be specified and download on-demand).

**Pull request closed without merging**